### ISSUE TITLE
Can we have ****_test.py discoverable as valid test file?

### ISSUE THREAD

**Issue by Praxkv1** [microsoft/nutter#43]
As of today this framework discovers the test methods if test cases are written into `test_*****.py` files. Can we enhance this to find the test cases if its written into `*****_test.py` as well?

**Comment by giventocode** [comment]
Thanks for the suggestion

**Comment by quanuw** [comment]
@giventocode Why does this method return False if search_pattern starts with "test_"?
```python
def _is_a_test_pattern(self, pattern):
        segments = pattern.split('/')
        if len(segments) > 0:
            search_pattern = segments[len(segments)-1]
            if search_pattern.lower().startswith('test_'):
                return False
            return True
        logging.Fatal(
            """ Invalid argument.
                 The value must be the full path to the test or a pattern """)
```

**Comment by giventocode** [comment]
The cli supports pattern matching for tests. If you provide:

```
nutter run /dev/foo*
```
The cli will look for test notebooks that have foo, e.g.  ```test_foo``` or ```test_foobar```

However, you can provide a full test notebook name, in which case the cli does not try apply pattern matching and instead assumes that you want run a specific test notebook. e.g. 

```
nutter run /dev/test_foo
```

This is the scenario where ```_is_a_test_pattern``` should return false.

**Comment by giventocode** [comment]
Fixed in 0.3.14
